src/Servers/Kestrel/Transport.MsQuic/src/Internal/MsQuicEnums.cs

```diff
@@ -0,0 +1,177 @@
+        CERT_FILE = 0x00000008,
+        ENABL_OCSP = 0x00000010,
+        CERT_NULL = 0xF0000000,
+    }
+
+    internal enum QUIC_LISTENER_EVENT : byte
+    {
+        NEW_CONNECTION = 0
+    }
+
+    internal enum QUIC_CONNECTION_EVENT : byte
+    {
+        CONNECTED = 0,
+        SHUTDOWN_BEGIN = 1,
+        SHUTDOWN_BEGIN_PEER = 2,
+        SHUTDOWN_COMPLETE = 3,
+        LOCAL_ADDR_CHANGED = 4,
+        PEER_ADDR_CHANGED = 5,
+        NEW_STREAM = 6,
+        STREAMS_AVAILABLE = 7,
+        PEER_NEEDS_STREAMS = 8,
+        IDEAL_SEND_BUFFER = 9,
+    }
+
+    [Flags]
+    internal enum QUIC_CONNECTION_SHUTDOWN : uint
+    {
+        NONE = 0x0,
+        SILENT = 0x1
+    }
+
+    internal enum QUIC_PARAM_REGISTRATION : uint
+    {
+        RETRY_MEMORY_PERCENT = 0,
+        CID_PREFIX = 1
+    }
+
+    internal enum QUIC_PARAM_LEVEL : uint
+    {
+        REGISTRATION = 0,
+        SESSION = 1,
+        LISTENER = 2,
+        CONNECTION = 3,
+        TLS = 4,
+        STREAM = 5,
+    }
+
+    internal enum QUIC_PARAM_LISTENER : uint
+    {
+        LOCAL_ADDRESS = 0
+    }
+
+    internal enum QUIC_PARAM_SESSION : uint
+    {
+        TLS_TICKET_KEY = 0,
+        PEER_BIDI_STREAM_COUNT = 1,
+        PEER_UNIDI_STREAM_COUNT = 2,
+        IDLE_TIMEOUT = 3,
+        DISCONNECT_TIMEOUT = 4,
+        MAX_BYTES_PER_KEY = 5
+    }
+
+    internal enum QUIC_PARAM_CONN : uint
+    {
+        QUIC_VERSION = 0,
+        LOCAL_ADDRESS = 1,
+        REMOTE_ADDRESS = 2,
+        IDLE_TIMEOUT = 3,
+        PEER_BIDI_STREAM_COUNT = 4,
+        PEER_UNIDI_STREAM_COUNT = 5,
+        LOCAL_BIDI_STREAM_COUNT = 6,
+        LOCAL_UNIDI_STREAM_COUNT = 7,
+        CLOSE_REASON_PHRASE = 8,
+        STATISTICS = 9,
+        STATISTICS_PLAT = 10,
+        CERT_VALIDATION_FLAGS = 11,
+        KEEP_ALIVE_ENABLED = 12,
+        KEEP_ALIVE_INTERVAL = 13,
+        DISCONNECT_TIMEOUT = 14,
+        SEC_CONFIG = 15,
+        USE_SEND_BUFFER = 16,
+        USE_PACING = 17,
+        SHARE_UDP_BINDING = 18,
+        IDEAL_PROCESSOR = 19,
+        MAX_STREAM_IDS = 20
+    }
+
+    internal enum QUIC_PARAM_STREAM : uint
+    {
+        ID = 0,
+        RECEIVE_ENABLED = 1,
+        ZERORTT_LENGTH = 2,
+        IDEAL_SEND_BUFFER = 3
+    }
+
+    [Flags]
+    internal enum QUIC_RECV_FLAG : byte
+    {
+        NONE = 0,
+        ZERO_RTT = 0x00000001
+    }
+
+    internal enum QUIC_STREAM_EVENT : byte
+    {
+        START_COMPLETE = 0,
+        RECV = 1,
+        SEND_COMPLETE = 2,
+        PEER_SEND_CLOSE = 3,
+        PEER_SEND_ABORT = 4,
+        PEER_RECV_ABORT = 5,
+        SEND_SHUTDOWN_COMPLETE = 6,
+        SHUTDOWN_COMPLETE = 7,
+        IDEAL_SEND_BUFFER_SIZE = 8,
+    }
+
+    [Flags]
+    internal enum QUIC_NEW_STREAM_FLAG : uint
+    {
+        NONE = 0,
+        UNIDIRECTIONAL = 0x1,
+        FAIL_BLOCKED = 0x2,
+    }
+
+    [Flags]
+    internal enum QUIC_STREAM_START : uint
+    {
+        NONE = 0,
+        FAIL_BLOCKED = 0x1,
+        IMMEDIATE = 0x2
```

Comment:
Can you make another pass through all these enums to make sure you aren't missing any. For instance, this one is missing `QUIC_STREAM_START_FLAG_ASYNC = 0x0004`.

Reply:
Done, a few other things needed to be updated. Good call.